packages-exp/auth-types-exp/index.d.ts

```diff
@@ -153,4 +153,53 @@ type ActionCodeOperationType =
  | 'EMAIL_SIGNIN'
  | 'VERIFY_EMAIL'
  | 'VERIFY_AND_CHANGE_EMAIL'
- | 'REVERT_SECOND_FACTOR_ADDITION'
\ No newline at end of file
+ | 'REVERT_SECOND_FACTOR_ADDITION'
+ 
+/**
+ * Supported providers
+ */
+export enum ProviderId {
+  ANONYMOUS = 'anonymous',
+  CUSTOM = 'custom',
+  FACEBOOK = 'facebook.com',
+  FIREBASE = 'firebase',
+  GITHUB = 'github.com',
+  GOOGLE = 'google.com',
+  PASSWORD = 'password',
+  PHONE = 'phone',
+  TWITTER = 'twitter.com'
+}
+
+/**
+ * Supported sign in methods
+ */
+export enum SignInMethod {
+  ANONYMOUS = 'anonymous',
+  EMAIL_LINK = 'emailLink',
+  EMAIL_PASSWORD = 'password',
+  FACEBOOK = 'facebook.com',
+  GITHUB = 'github.com',
+  GOOGLE = 'google.com',
+  PHONE = 'phone',
+  TWITTER = 'twitter.com'
+}
+ 
+export interface AuthCredential {
+  readonly providerId: ProviderId;
+  readonly signInMethod: SignInMethod;
+  toJSON(): object;
+  fromJSON(json: object | string): AuthCredential | null;
+}
+
+export class OAuthCredential implements AuthCredential {
```

Comment:
this will have to be "export declare class" since code can't live in .d.ts

Reply:
this seems to work since it's an abstract class

Reply:
It's not declared `abstract` though?

Reply:
done